ms-engine-spring-jaxrs/src/main/java/net/trajano/ms/engine/jaxrs/JaxRsRouter.java

```diff
@@ -0,0 +1,103 @@
+        final Router router,
+        final Handler<RoutingContext> jaxRsHandler) {
+
+        final Reflections reflections = new Reflections(applicationClass);
+        final String rootPath = Optional.ofNullable(applicationClass.getAnnotation(ApplicationPath.class)).map(ApplicationPath::value).orElse("/");
+
+        final SortedSet<JaxRsPath> paths = new TreeSet<>();
+        reflections.getTypesAnnotatedWith(Path.class).forEach(clazz -> {
+            final String classPath = Optional.ofNullable(clazz.getAnnotation(Path.class)).map(Path::value).orElse("");
+            stream(clazz.getMethods()).filter(m -> m.getAnnotation(GET.class) != null
+                || m.getAnnotation(POST.class) != null
+                || m.getAnnotation(PUT.class) != null
+                || m.getAnnotation(DELETE.class) != null).forEach(m -> {
+                    final String path = Optional.ofNullable(m.getAnnotation(Path.class)).map(Path::value).orElse("");
+                    final String[] consumes = Optional.ofNullable(m.getAnnotation(Consumes.class)).map(Consumes::value).orElse(new String[0]);
+                    final String[] produces = Optional.ofNullable(m.getAnnotation(Produces.class)).map(Produces::value).orElse(new String[0]);
+
+                    final boolean get = m.getAnnotation(GET.class) != null;
+                    final boolean post = m.getAnnotation(POST.class) != null;
+                    final boolean put = m.getAnnotation(PUT.class) != null;
+                    final boolean delete = m.getAnnotation(DELETE.class) != null;
+                    paths.add(new JaxRsPath(rootPath + classPath + path, consumes, produces, get, post, put, delete));
+                });
+        });
+        paths.stream().filter(p -> !p.isGet()).forEach(p -> stream(p.getConsumes()).forEach(consumes -> stream(p.getProduces()).forEach(produces -> {
+            final Route route;
+            if (p.isPost()) {
+                if (p.isExact()) {
+                    route = router.post(p.getPath());
+                } else {
+                    route = router.postWithRegex(p.getPathRegex());
+                }
+            } else if (p.isPut()) {
+                if (p.isExact()) {
+                    route = router.put(p.getPath());
+                } else {
+                    route = router.putWithRegex(p.getPathRegex());
+                }
+            } else if (p.isDelete()) {
+                if (p.isExact()) {
+                    route = router.delete(p.getPath());
+                } else {
+                    route = router.deleteWithRegex(p.getPathRegex());
+                }
+            } else {
+                throw new IllegalStateException("JaxRsPath=" + p + " does not have a HTTP Method active");
+            }
+            route.consumes(consumes).produces(produces).handler(jaxRsHandler);
+
+        })));
+        paths.stream().filter(p -> p.isGet()).forEach(p -> stream(p.getProduces()).forEach(produces -> {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this lambda with a method reference. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1612)